packages/core/src/shapes/Shapes.ts

```diff
@@ -185,6 +188,44 @@ const Text = ShapeDef({
   isPolygonlike: true,
 });
 
+// bboxFromGroup is here, not in engine/BBox, to prevent cyclic import
```

Comment:
I thought JavaScript was generally OK with cyclic imports; could you clarify what exactly breaks when you try to put this in `engine/BBox`?